integration_tests/src/main/python/asserts.py

```diff
@@ -557,13 +559,23 @@ def run_on_gpu():
 
     return (from_cpu, from_gpu)
 
-def assert_gpu_and_cpu_are_equal_collect(func, conf={}, is_cpu_first=True):
+def assert_gpu_and_cpu_are_equal_collect(func, conf={}, is_cpu_first=True, result_converter_func_before_compare=None):
```

Comment:
nit: How about give a shorter name, e.g. `post_handle_result`. Since the description covers its details.

Reply:
Let's use `result_canonicalize_func_before_compare`, actually developer may not look at the description, and the parameter name has a good meaning to let developer to know the usage.